app/models/edition.rb

```diff
@@ -161,23 +129,25 @@ def assign_status(state,
     attributes = { status: status }
 
     if update_last_edited
-      attributes[:last_edited_at] = Time.current
-      attributes[:last_edited_by] = user
+      assign_as_edit(user, attributes)
+    else
+      assign_attributes(attributes)
+      self
     end
+  end
 
-    assign_attributes(attributes)
+  def assign_as_edit(user, attributes)
```

Comment:
I didn't find this the easiest method name to understand, but I don't have a better suggestion. Seems good to have it encapsulated in a method though 👍

Reply:
Agreed - it's simply the first thing that came to mind.